### PR TITLE
Force timeout during connection

### PR DESCRIPTION
**Pull request by jayvdb**
Fixes https://github.com/tailscale/github-action/issues/134

### CODE REVIEW

**Comment by jayvdb** [comment]
ping @knyar & @patrickod & @willnorris

File: action.yml [hunk 1]
```diff
@@ -44,6 +44,10 @@ inputs:
     description: 'Optional state directory to use (if unset, memory state is used)'
     required: false
     default: ''
+  timeout:
```
Comment (willnorris) [inline action.yml:47]: I'd rather not expand the API surface of the action if we really don't need to.  Users can already set the timeout on either the entire job or the step, so I'm not sure that having a configurable timeout is necessary here also.

File: action.yml [hunk 2]
```diff
@@ -127,4 +132,4 @@ runs:
             TAILSCALE_AUTHKEY="${{ inputs['oauth-secret'] }}?preauthorized=true&ephemeral=true"
             TAGS_ARG="--advertise-tags=${{ inputs.tags }}"
           fi
-          timeout 5m sudo -E tailscale up ${TAGS_ARG} --authkey=${TAILSCALE_AUTHKEY} --hostname=${HOSTNAME} --accept-routes ${ADDITIONAL_ARGS}
+          timeout --verbose --kill-after=1s ${TIMEOUT} sudo -E tailscale up ${TAGS_ARG} --authkey=${TAILSCALE_AUTHKEY} --hostname=${HOSTNAME} --accept-routes ${ADDITIONAL_ARGS}
```
Comment (willnorris) [inline action.yml:135]: ooh, nice.  I hadn't used `--kill-after` before.  Definitely makes sense here if this timeout isn't taking effect.

File: action.yml (hunk 1 above)
Comment (willnorris) [reply action.yml:47]: oh, I just noticed that the original issue this was fixing was specifically to add a configurable timeout.  I'm still curious if the step timeout does not also accomplish the same thing?

File: action.yml (hunk 2 above)
Comment (jayvdb) [reply action.yml:135]: Correct, the timeout isnt working at all.
Please see https://github.com/tailscale/github-action/issues/134

**Review by willnorris — APPROVED** [review]

**Pull request merged**